tools/cov2json/ProfileData.hpp

```diff
@@ -0,0 +1,121 @@
+//===----- ProfileData.hpp - Data structures for serializing profdata -----===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef ProfileData_hpp
+#define ProfileData_hpp
+
+#include <stdio.h>
+#include <iostream>
+#include "llvm/ProfileData/InstrProfReader.h"
+#include "llvm/ProfileData/CoverageMapping.h"
+#include "llvm/ProfileData/CoverageMappingReader.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/FormattedStream.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/Path.h"
+#include <map>
+#include <set>
+
+namespace llvm {
+namespace cov2json {
+
+struct Region {
+public:
+  unsigned columnStart, columnEnd, lineStart, lineEnd;
+  uint64_t executionCount;
+
+  Region(unsigned columnStart, unsigned columnEnd, unsigned lineStart,
+         unsigned lineEnd, uint64_t executionCount)
+      : columnStart(columnStart), columnEnd(columnEnd), lineStart(lineStart),
+        lineEnd(lineEnd), executionCount(executionCount) {}
+  Region(llvm::coverage::CountedRegion &region)
+      : Region(region.ColumnStart, region.ColumnEnd, region.LineStart,
+               region.LineEnd, region.ExecutionCount) {}
+  Region() {}
+};
+
+struct Function {
+public:
+  std::string name;
+  std::string symbol;
+  std::vector<Region> regions;
+  uint64_t executionCount;
+  Function(StringRef name, std::vector<Region> regions,
+           uint64_t executionCount)
+      : name(name), regions(regions), executionCount(executionCount) {}
+
+  uint64_t regionCount();
+  uint64_t executedRegionCount();
+  
+  Function(const llvm::coverage::FunctionRecord &record);
+
+  Function(const Function &copy)
+      : symbol(copy.symbol), name(copy.name), regions(copy.regions),
+        executionCount(copy.executionCount) {}
+
+  Function(Function &copy)
+      : symbol(copy.symbol), name(copy.name), regions(copy.regions),
+        executionCount(copy.executionCount) {}
+
+  Function() {}
+};
+
+/// A struct that stores all functions associated with a given source file.
+struct File {
+public:
+  std::string name;
+  std::vector<Function> functions;
+  
+  uint64_t regionCount();
+  uint64_t executedRegionCount();
+
+  File(StringRef name, std::vector<Function> functions)
+      : name(name), functions(functions) {}
+
+  File(File &copy)
```

Comment:
You shouldn't need this constructor since you have `File(const File &copy)` below.